app/views.py

```diff
@@ -0,0 +1,160 @@
+from app.forms import RegistrationForm
+from app.forms import EditProfileForm, ProfilePasswordChangeForm
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth import logout
+from django.views.decorators.csrf import csrf_protect
+from django.shortcuts import render_to_response
+from django.http import HttpResponseRedirect
+from django.template import RequestContext
+from app.models import UserProfile
+import datetime
+# from django import forms
+from django.views.generic.base import TemplateView
+from django.views.generic.edit import FormView
+from django.core.urlresolvers import reverse
+# from django.shortcuts import render
+from django.contrib import messages
+from django.contrib.auth.models import User
+
+
+@csrf_protect
+def register(request):
+    if request.method == 'POST':
+        form = RegistrationForm(request.POST)
+        if form.is_valid():
+            try:
+                date = datetime.datetime(
+                    year=int(form.cleaned_data['year']),
+                    month=int(form.cleaned_data['month']),
+                    day=int(form.cleaned_data['day'])
+                )
+                dob = date.strftime('%Y-%m-%d')
+                user = User.objects.create_user(
+                    username=form.cleaned_data['username'],
+                    password=form.cleaned_data['password1'],
+                )
+                profile = UserProfile(user=user, date_of_birth=dob)
+                profile.save()
+                return HttpResponseRedirect('/')
+            except ValueError:
+                messages.error(request, 'The day is out of range for month.')
```

Comment:
Please put this `try`...`except` block only around the piece that can reasonable raise that error, which I think is the `datetime()` call on line 26.